### PR TITLE
fix: android build

### LINKED ISSUE #91: fatal error: 'tensorflow/lite/c/c_api.h' file not found
on 1.4.0, I have the following error on mac m2 when compiling:

In file included from /Users/pingou/react/app/node_modules/react-native-fast-tflite/cpp/TensorHelpers.cpp:9:
/Users/pingou/react/app/node_modules/react-native-fast-tflite/android/../cpp/TensorHelpers.h:15:10: fatal error: 'tensorflow/lite/c/c_api.h' file not found
#include <tensorflow/lite/c/c_api.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~

### PR DESCRIPTION
**Pull request by bang9**
### Summary

- **Compatibility**: Supports React Native merged `.so` changes introduced in version 0.76.  
- **LiteRT Migration**: Adopts LiteRT(It's new name for TensorFlow), updating to [LiteRT v1.0.1](https://github.com/google-ai-edge/LiteRT/releases/tag/v1.0.1), which includes TensorFlow 2.17.0.  
  - Resolves header issues in the Android build process. (Closes #91, #99, #118)  
  - Ensures alignment with the same TensorFlow version used on iOS.  
- **Submodule Removal**: Eliminates the submodule as pre-building is no longer required with the updated setup.

### CODE REVIEW

**Review by lucksp — APPROVED** [review]
you're a savior for doing this!   I was able to run on my Android device...might want 1 or 2 other approvals.

Note: you may need to update vision-camera-resize-plugin

**Comment by icarv** [comment]
awesome 🎉  these changes are working for me in expo 52 building locally

**Review by RayanMoarkech — COMMENTED** [review]
If submodules are not needed anymore, then pay attention to delete the scripts in package.json that uses submodules:
```
git submodule
```

File: android/CMakeLists.txt [hunk 1]
```diff
@@ -42,7 +42,7 @@ target_include_directories(
   PRIVATE
   "../cpp"
   "src/main/cpp"
-  "../tensorflow/"
+  "src/main/cpp/lib/litert/headers"
```
Comment (mrousavy) [inline android/CMakeLists.txt:45]: does it really extract here?

**Review by lucksp — COMMENTED** [review]
can `litert` be used in iOS too?

**Review by lucksp — COMMENTED** [review]
sorry, looking at this more - for Android, does anything need to happen for the GPU delegates, as [suggested in docs](https://ai.google.dev/edge/litert/android/gpu)?

**Review by RayanMoarkech — APPROVED** [review]

**Comment by kuldip-simform** [comment]
Thank you for making this PR. If you have any ETA in mind for this PR to merge and the new version release date, can you please let us know?
Currently, this is a blocker for us.

This will be much appreciated. Thank you.

**Comment by mrousavy** [comment]
Let's go. Thank you!

**Pull request merged**